src/main/java/org/mozilla/zest/core/v1/ZestJSON.java

```diff
@@ -81,33 +83,39 @@ private static GsonBuilder newDefaultGsonBuilder() {
                 .setPrettyPrinting();
     }
 
-    @Override
-    public ZestElement deserialize(
-            JsonElement element, Type rawType, JsonDeserializationContext arg2) {
-        if (element instanceof JsonObject) {
-            String elementType = ((JsonObject) element).get("elementType").getAsString();
-
-            if (elementType.startsWith("Zest")) {
-                try {
-                    // Class<?> c =
-                    // Class.forName(this.getClass().getPackage().getName()"org.mozilla.zest.core.v1." + elementType);
-                    Class<?> c =
-                            Class.forName(
-                                    this.getClass().getPackage().getName() + "." + elementType);
-                    return (ZestElement) getGson().fromJson(element, c);
-
-                } catch (ClassNotFoundException e) {
-                    throw new JsonParseException(e);
+    private static class ZestTypeAdapter
+            implements JsonDeserializer<ZestElement>, JsonSerializer<ZestElement> {
+
+        static final ZestTypeAdapter INSTANCE = new ZestTypeAdapter();
+
+        @Override
+        public ZestElement deserialize(
+                JsonElement element, Type rawType, JsonDeserializationContext arg2) {
+            if (element instanceof JsonObject) {
+                String elementType = ((JsonObject) element).get("elementType").getAsString();
+
+                if (elementType.startsWith("Zest")) {
+                    try {
+                        // Class<?> c =
```

Comment:
Comment no longer needed?

Reply:
Removed and extracted a constant for the package.